src/wp-includes/functions.php

```diff
@@ -7635,3 +7633,72 @@ function is_php_version_compatible( $required ) {
 function wp_fuzzy_number_match( $expected, $actual, $precision = 1 ) {
 	return abs( (float) $expected - (float) $actual ) <= $precision;
 }
+
+/**
+ * Wraps inline JavaScript in `<script>` tags.
+ *
+ * Enables to inject attributes in the `<script>` tag via the `script_attributes` filter.
+ * Automatically injects type attribute if needed.
+ *
+ * @param array     $attributes `<script>` tag attributes.
+ * @param bool      $double_quotes  Optional. Uses double quotes `"` to wrap attribute values if true, single quotes `'` otherwise.
```

Comment:
Single quotes are used in every script tag generated by WP_Scripts.
I included the option because script tags are sometimes added to the page through document.write; depending on which type of quote is used, the other needs to be used to wrap the attributes' values.
Example of the above: https://github.com/WordPress/wordpress-develop/blob/a85d190b00564d6983e1155909f0eb70ae9e41b6/src/wp-includes/script-loader.php#L205-L216
I added this parameter to make refactoring more straightforward, but I see it doesn't seem to bring as many benefits as drawbacks.